homeassistant/components/arcam_fmj/config_flow.py

```diff
@@ -16,12 +27,67 @@ class ArcamFmjFlowHandler(config_entries.ConfigFlow):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
-    async def async_step_import(self, import_config):
-        """Import a config entry from configuration.yaml."""
-        entries = self.hass.config_entries.async_entries(DOMAIN)
-        import_key = _GETKEY(import_config)
-        for entry in entries:
-            if _GETKEY(entry.data) == import_key:
-                return self.async_abort(reason="already_setup")
+    async def _async_set_unique_id_and_update(self, host, port, uuid):
+        await self.async_set_unique_id(uuid)
+        self._abort_if_unique_id_configured({CONF_HOST: host, CONF_PORT: port})
+
+    def _async_create_standard_entry(self, host, port):
+        return self.async_create_entry(
+            title=f"{DEFAULT_NAME} ({host})", data={CONF_HOST: host, CONF_PORT: port},
+        )
+
+    async def async_step_user(self, user_info):
+        """Handle a discovered device."""
+        errors = {}
+
+        if user_info is not None:
+            uuid = await get_uniqueid_from_host(
```

Comment:
What happens if the connection here fails? We want to check the connection works before we create the entry.

Reply:
Added start test